### ISSUE TITLE
Improvement Request: SimpleDateFormat in Jackson ObjectMapper

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#121]
This line: 

```java
        objectMapper.setDateFormat(new SimpleDateFormat("yyyy-MM-dd")); 
```

is flagged with `AvoidSimpleDateFormat`. But objectMapper makes sure in this case that the usage is thread safe by making clones. See: https://stackoverflow.com/questions/3907929/should-i-declare-jacksons-objectmapper-as-a-static-field#comment-26348209

**Comment by jborgers** [comment]
Also holds for XMLMapper which is a subclass of ObjectMapper.

**Comment by jborgers** [comment]
I checked the source code and the clone is actually taken, including a clone of the fields for DateFormat: Calendar and NumberFormat.

**Comment by jborgers** [comment]
However, a clone will make sure it is not shared with the caller, yet in the following case, after cloning I suspect the dateformat will be used by multiple threads:

```
    private static final XmlMapper XML_MAPPER;
    static{
        XML_MAPPER = new XmlMapper();
        @SuppressWarnings("PMD.AvoidSimpleDateFormat") // Since XMLMapper expects SimpleDateFormat as the parameter. So It is suppressed.
        SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss"); //NOSONAR
        XML_MAPPER.setDateFormat(df);
        XML_MAPPER.setSerializationInclusion(JsonInclude.Include.NON_NULL);
    }
```

**Comment by jborgers** [comment]
Some more digging shows that where DateFormats are used with parse and format, thread-safety is taken into account and appears to be thread-safe.

**Comment by stokpop** [comment]
Can this be closed?

**Comment by jborgers** [comment]
yes
